src/driver.rs

```diff
@@ -182,6 +191,45 @@ impl rustc_driver::Callbacks for ClippyCallbacks {
     }
 }
 
+fn build_disabled_set(sess: &Session, conf: &'static Conf) -> FxHashSet<&'static str> {
+    if sess.is_test_crate() {
+        return FxHashSet::default();
+    }
```

Comment:
For my own understanding, shouldn't this condition be inverted? The idea AFAIK is to essentially ignore `disabled-in-tests` and return an empty set when *not* in a test crate, that is, we don't want to register this expect hook when not in a test crate, right?

Reply:
Oof. You're right.